crowbar_framework/app/models/service_object.rb

```diff
@@ -1540,6 +1524,79 @@ def wait_for_chef_daemons(node_list)
 
   private
 
+  THREAD_POOL_SIZE = 20
+
+  def release_chef_locks(locks)
+    return if locks.empty?
+
+    queue = Queue.new
+    locks.each { |l| queue.push l }
+
+    workers = (1...THREAD_POOL_SIZE).map do
```

Comment:
@aplanas can you explain this a bit further? 

I kind of understand what you mean here but not sure, so i'll be better to clarify, but Im guessing its that the queue could be >20 but we only create 20 threads max, so in a case where we deploy to 25 nodes, 5 nodes will not get a lock and be stuck in the queue, which could be fixed by working in 20 threads batches but doing it `unless queue.empty?`

Am i right here?